examples/multilang/mycomponent/nodejs/index.ts

```diff
@@ -0,0 +1,54 @@
+// This file would be autogenerated from schema.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+export interface MyComponentArgs {
+    input1: pulumi.Input<number>;
+}
+
+export class MyComponent extends pulumi.remote.ProxyComponentResource {
+    public myid!: pulumi.Output<string>;
+    public output1!: pulumi.Output<number>;
+    public innerComponent!: MyInnerComponent;
+    public nodeSecurityGroup!: aws.ec2.SecurityGroup;
+    constructor(name: string, args: MyComponentArgs, opts?: pulumi.ComponentResourceOptions) {
+        super(
+            "my:mod:MyComponent",
+            name,
+            require.resolve(".."),
+            "MyComponent",
+            args,
+            {
+                myid: undefined,
+                output1: undefined,
+                innerComponent: undefined,
+                nodeSecurityGroup: undefined,
+            },
+            opts,
+        );
+    }
+}
+pulumi.runtime.registerProxyConstructor("my:mod:MyComponent", MyComponent);
+
+
+export interface MyInnerComponentArgs {
+}
+
+export class MyInnerComponent extends pulumi.remote.ProxyComponentResource {
+    public data!: pulumi.Output<string>;
+    constructor(name: string, args: MyInnerComponentArgs, opts?: pulumi.ComponentResourceOptions) {
+        super(
+            "my:mod:MyInnerComponent",
+            name,
+            require.resolve(".."),
+            "MyInnerComponent",
```

Comment:
1. is there a relation between this name and the type?  if so, we might consider making it computed instead of required.  
2. is the `require.resolve` necessary?  or can it be figured out by the base type?  Is the purpose to pass along the path where this code lives so it can be loaded at runtime?

Reply:
> is there a relation between this name and the type? if so, we might consider making it computed instead of required.

Not necessarily - the type is a Pulumi type token, the `"MyInnerComponent"` here is the name of a JavaScript property on the module object resolved above.  We will likely have style guidance that these are the same - but I suspect we won't want to strictly tie them together as they are logically quite different things.

> is the require.resolve necessary? or can it be figured out by the base type? Is the purpose to pass along the path where this code lives so it can be loaded at runtime?

It is needed to compute the path relative to the module this runs in.   This is just to get an absolute path to the file for distribution.   The whole structure of this layout likely will be different for any "real" package (still a bit TBD), but something like this will probably be needed to locate the Node package.